src/main/java/com/synopsys/integration/detect/ApplicationUpdater.java

```diff
@@ -197,61 +257,55 @@ private boolean runMainClass(Path jarPath, String[] launchArgs)
                     while (jarEntryEnum.hasMoreElements()) {
                         final JarEntry jarEntry = jarEntryEnum.nextElement();
                         String jarEntryName = jarEntry.getName();
                         if (jarEntry.getName().startsWith("org/springframework/boot")  
-                                && jarEntry.getName().endsWith(".class") == true) {
+                                && jarEntry.getName().endsWith(".class")) {
                             int endIndex = jarEntryName.lastIndexOf(".class");
                             String className = jarEntryName.substring(0, endIndex).replace('/', '.');
                             final Class<?> loadedClass = urlClassLoader.loadClass(className);
                             classMap.put(loadedClass.getName(), loadedClass);
                         }
                     }
-                    urlClassLoader.close();
                 }
-                jarFile.close();
             }
             final Class<?> jarFileArchiveClass = classMap.get("org.springframework.boot.loader.archive.JarFileArchive");
             final Constructor<?> jarFileArchiveConstructor = jarFileArchiveClass.getConstructor(File.class);
             final Object jarFileArchive = jarFileArchiveConstructor.newInstance(new File(pathToJar));
             final Class<?> archiveClass = classMap.get("org.springframework.boot.loader.archive.Archive");
-            final Class mainClass = classMap.get("org.springframework.boot.loader.JarLauncher");
+            final Class<?> mainClass = classMap.get("org.springframework.boot.loader.JarLauncher");
             final Constructor<?> jarLauncherConstructor = mainClass.getDeclaredConstructor(archiveClass);
-            jarLauncherConstructor.setAccessible(true);
             final Object jarLauncher = jarLauncherConstructor.newInstance(jarFileArchive);
             final Class<?> launcherClass = 	classMap.get("org.springframework.boot.loader.Launcher");
             final Method launchMethod = launcherClass.getDeclaredMethod("launch", String[].class);
-            launchMethod.setAccessible(true);
             checkEnvironmentProperties();
             args = parseArguments(args);
-            launchMethod.invoke(jarLauncher, new Object[]{args});
+            launchMethod.invoke(jarLauncher, (Object[]) args);
         } finally {
             close();
         }
         return true;
     }
     
-    private boolean checkInstallationDir(Path path) throws NoSuchFileException, AccessDeniedException, IOException {
+    private boolean checkInstallationDir(Path path) throws IOException {
         final Path parentPath = path.getParent();
         final Path directoryPath = Files.createDirectories(parentPath);
         
-        if (!Files.exists(directoryPath)) {
+        if (!directoryPath.toFile().exists()) {
             String parentPathString = (parentPath != null) ? parentPath.toString() : null;
```

Comment:
Nit: Since directoryPath exists & it is created from parent path, `parentPath != null` check here might be redundant (if checking before `createDirectories`). Doesn't hurt to have it though

Reply:
parentPath won't be null because path here is the installationDirectory which is already checked and confirmed to exist.